crates/boundless-market/src/contracts/boundless_market.rs

```diff
@@ -1381,6 +1392,133 @@ pub fn decode_calldata(data: &Bytes) -> Result<(Vec<Fulfillment>, AssessorReceip
     ))
 }
 
+#[derive(Debug, Clone)]
+/// Represents the parameters for submitting a Merkle Root.
+pub struct Root {
+    /// The address of the set verifier contract.
+    pub verifier_address: Address,
+    /// The Merkle root of the proof.
+    pub root: B256,
+    /// The seal of the proof.
+    pub seal: Bytes,
+}
+
+#[derive(Debug, Clone)]
+/// Represents the parameters for pricing a request.
+pub struct Price {
+    /// The list of requests to be priced.
+    pub requests: Vec<ProofRequest>,
+    /// The list of client signatures for the requests.
+    pub client_sigs: Vec<Bytes>,
+}
```

Comment:
I would avoid calling it price. I think the fact that this appears in the public API is a bit unfortunate and I would avoid it if we can. I would prefer to this as an "unlocked request" instead. I would also refactor this to be a single `UnlockedRequest { request: ProofRequest, client_signature: Bytes }` and then wrap that in `Vec<UnlockedRequest>` on the `FulfillmentBuilder`. We can easily unzip this struct when it comes to to build the calldata.